vault.js

```diff
@@ -1,4 +1,27 @@
 'use strict';
-module.exports = function() {
-  
-};
\ No newline at end of file
+function vaultGen(){
+  var secretValue = {};
+
+  function getValue (key){
+    if (!key){
+      return null;
+    }else if(!secretValue.hasOwnProperty(key)){
```

Comment:
You are testing `key` twice. Think about your primary functionality. if you have a `key` you can return it otherwise return `null`
Simplify this.